install.sh

```diff
@@ -543,6 +543,17 @@ EOABORT
 )"
 fi
 
+# Multi-user support on macOS
+directories=("$HOMEBREW_REPOSITORY" "$HOMEBREW_CACHE")
+if [[ -z "${HOMEBREW_ON_LINUX-}" ]] && [[ -n "${HOMEBREW_ENABLE_MULTI_USER-}" ]]; then
+  for dir in "${directories[@]}"; do
+    if [[ -d "${dir}" ]]; then
+      execute_sudo "$CHGRP" "-R" "admin" "${dir}"/*
+      execute_sudo "/bin/chmod" "-R" "g+rwx" "${dir}"/*
```

Comment:
It'd be good to make this part of the `group_chmods` and `chgrps` logic above. I think this could be done by using the `file_not_grpowned` and perhaps `exists_but_not_writable` above. Similarly, on `The following existing directories will be made group writable` it'd be nice to e.g. advertise the `HOMEBREW_ENABLE_MULTI_USER` (perhaps renamed to `HOMEBREW_MULTI_USER`) variable and tell people they can Ctrl-C and rerun with it set in order to use as multiple `$GROUP` users.

Reply:
@MikeMcQuaid  A cleaner implementation, thanks! Updated the code accordingly